examples/serialization/avro/avro-cli.py

```diff
@@ -0,0 +1,180 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+
+from six.moves import input
+
+from confluent_kafka import avro as avro
+
+record_schema_str = """
+    {
+        "type": "record",
+        "name": "User",
+        "fields": [
+            {"name": "name", "type": "string"},
+            {"name": "favorite_number",  "type": ["int", "null"]},
+            {"name": "favorite_color", "type": ["string", "null"]}
+        ]
+    }
+"""
+
+
+class ParserOpts(object):
+    def _build(self):
+        conf = {'bootstrap.servers': self.bootstrap_servers,
+                'schema.registry.url': self.schema_registry}
+
+        if self.userinfo:
+            conf['schema.registry.basic.auth.credentials.source'] = 'USER_INFO'
+            conf['schema.registry.basic.auth.user.info'] = self.userinfo
+        return conf
+
+    def producer_conf(self):
+        return self._build()
+
+    def consumer_conf(self):
+        return dict({"group.id": self.group}, **self._build())
+
+
+class User(dict):
+    _schema = avro.loads(record_schema_str)
+
+    def __init__(self, conf=None):
+        if not conf:
+            conf = {"name": "anonymous",
+                    "favorite_number": 0,
+                    "favorite_color": ""}
+        super(User, self).__init__(conf)
+
+    @classmethod
+    def schema(cls):
+        return cls._schema
+
+    def prompt(self):
+        self['name'] = input("Enter name:")
+        num = input("Enter favorite number:")
+        if num == '':
+            num = 0
+        self['favorite_number'] = int(num)
+        self['favorite_color'] = input("Enter favorite color:")
+
+
+def on_delivery(err, msg):
+    if err is not None:
+        print('Message delivery failed ({} [{}]): %{}'.format(msg.topic(), str(msg.partition()), err))
+        return 0
+    elif err is not None:
+        print('Message delivery failed {}'.format(err))
+        return 0
+    else:
+        print('Message delivered to {} [{}] at offset [{}]: {}'.format(
+              msg.topic(), msg.partition(), msg.offset(), msg.value()))
+        return 1
+
+
+def produce(args):
+    from confluent_kafka.avro import AvroProducer
+
+    record = User()
+    topic = args.topic
+    conf = args.producer_conf()
+
+    producer = AvroProducer(conf, default_value_schema=record.schema())
+
+    print("producing user records to topic {}. ^c to exit.".format(topic))
+
+    while True:
+        try:
+            record.prompt()
```

Comment:
Suggest instantiating a new User object for each loop, reusing the same object for multiple Users seems weird to me and has problems if the object is passed to delivery report.